content/development/contribution_guidelines.md

```diff
@@ -109,97 +109,118 @@ The older style of representing a backslash using the character class `[\\\\]` m
 ```python
 "@rx a|(b|c)d"
 ```
-This will create a non-capturing group, guaranteeing the precedence of the alternative without capturing the input:
+
+On the other hand, this will create a _non-capturing_ group, guaranteeing the precedence of the alternative _without_ capturing the input:
+
 ```python
 "@rx a|(?:b|c)d"
 ```
 
 ### Lazy Matching
 
-The question mark `?` can be used to turn "greedy" quantifiers into "lazy" quantifiers, i.e., `.+` and `.*` are greedy while `.+?` and `.*?` are lazy. Using lazy quantifiers can help you write certain expressions that wouldn't otherwise be possible. However, in backtracking regular expression engines like PCRE, lazy quantifiers can also be a source of performance issues. The following is an example of an expression that uses a lazy qantifier:
+The question mark `?` can be used to turn "greedy" quantifiers into "lazy" quantifiers, i.e. `.+` and `.*` are greedy while `.+?` and `.*?` are lazy. Using lazy quantifiers can help with writing certain expressions that wouldn't otherwise be possible. However, in backtracking regular expression engines, like PCRE, lazy quantifiers can also be a source of performance issues. The following is an example of an expression that uses a lazy quantifier:
+
 ```python
 "@rx (?i)\.cookie\b.*?;\W*?(?:expires|domain)\W*?="
 ```
+
 This expression matches cookie values in HTML to detect session fixation attacks. The input string could be `document.cookie = "name=evil; domain=https://example.com";`.
 
-The lazy quantifiers in this expression are used to reduce the amount of backtracking that engines such as PCRE have to perform (others, such as re2, are not affected by this). Since the asterisk `*` is greedy, `.*` would match every character in the input up to the end, at which point the regular expression engine would realize that the next character, `;`, can't be matched and it will backtrack to the previous position (`;`). A few iterations later, the engine will realize that the character `d` from `domain` can't be matched and it will backtrack again. This will happen again and again, until the `;` at `evil;` if found. Only then can the engine proceed with the next expression part.
+The lazy quantifiers in this expression are used to reduce the amount of backtracking that engines such as PCRE have to perform (others, such as RE2, are not affected by this). Since the asterisk `*` is greedy, `.*` would match every character in the input up to the end, at which point the regular expression engine would realize that the next character, `;`, can't be matched and it will backtrack to the previous position (`;`). A few iterations later, the engine will realize that the character `d` from `domain` can't be matched and it will backtrack again. This will happen again and again, until the `;` at `evil;`, if found. Only then can the engine proceed with the next part of the expression.
```

Comment:
```suggestion
The lazy quantifiers in this expression are used to reduce the amount of backtracking that engines such as PCRE have to perform (others, such as RE2, are not affected by this). Since the asterisk `*` is greedy, `.*` would match every character in the input up to the end, at which point the regular expression engine would realize that the next character, `;`, can't be matched and it will backtrack to the previous position (`;`). A few iterations later, the engine will realize that the character `d` from `domain` can't be matched and it will backtrack again. This will happen again and again, until the `;` at `evil;` is found. Only then can the engine proceed with the next part of the expression.
```

Reply:
Ah! That's what the meaning was :) Corrected in new commit.